### PR TITLE
Bael 3596 asynchronous method calls in java

### PR DESCRIPTION
**Pull request by SmartyAnsh**

### CODE REVIEW

File: core-java-modules/core-java-concurrency-advanced-3/src/main/java/com/baeldung/async/JavaAsync.java [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+        AsyncCallable<Long> asyncCallable = Callables.asAsyncCallable(new Callable<Long>() {
+            public Long call() {
+                return factorial(number);
+            }
+        }, service);
+        return Futures.submitAsync(asyncCallable, service);
+    }
+
+    /**
+     *
```
Comment (eric-martin) [inline core-java-modules/core-java-concurrency-advanced-3/src/main/java/com/baeldung/async/JavaAsync.java]: remove this extra line

File: core-java-modules/core-java-concurrency-advanced-3/src/main/java/com/baeldung/async/JavaAsync.java [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+     */
+    @Loggable
+    public static long factorialUsingEAAsync(int number) {
+        CompletableFuture<Long> completableFuture = CompletableFuture.supplyAsync(() -> factorial(number));
+        long result = await(completableFuture);
+        return result;
+    }
+
+    /**
+     *
```
Comment (eric-martin) [inline core-java-modules/core-java-concurrency-advanced-3/src/main/java/com/baeldung/async/JavaAsync.java]: remove this extra line

File: core-java-modules/core-java-concurrency-advanced-3/src/main/java/com/baeldung/async/JavaAsync.java (hunk 1 above)
Comment (SmartyAnsh) [reply core-java-modules/core-java-concurrency-advanced-3/src/main/java/com/baeldung/async/JavaAsync.java]: fixed

File: core-java-modules/core-java-concurrency-advanced-3/src/main/java/com/baeldung/async/JavaAsync.java (hunk 2 above)
Comment (SmartyAnsh) [reply core-java-modules/core-java-concurrency-advanced-3/src/main/java/com/baeldung/async/JavaAsync.java]: fixed

File: core-java-modules/core-java-concurrency-advanced-3/pom.xml [hunk 3]
```diff
@@ -23,6 +23,37 @@
...
+            <groupId>org.aspectj</groupId>
+            <artifactId>aspectjrt</artifactId>
+            <version>1.9.5</version>
+            <scope>runtime</scope>
+        </dependency>
+        
+        <dependency>
+            <groupId>com.google.guava</groupId>
+            <artifactId>guava</artifactId>
+            <version>28.1-jre</version>
```
Comment (KevinGilmore) [inline core-java-modules/core-java-concurrency-advanced-3/pom.xml]: 28.2-jre is now available

File: core-java-modules/core-java-concurrency-advanced-3/pom.xml [hunk 4]
```diff
@@ -23,6 +23,37 @@
...
+        <dependency>
+            <groupId>org.cactoos</groupId>
+            <artifactId>cactoos</artifactId>
+            <version>0.43</version>
+        </dependency>
+
+        <dependency>
+            <groupId>com.ea.async</groupId>
+            <artifactId>ea-async</artifactId>
+            <version>1.2.3</version>
```
[the remark this answers is not in the document]
Comment (SmartyAnsh) [reply core-java-modules/core-java-concurrency-advanced-3/pom.xml]: fixed

File: core-java-modules/core-java-concurrency-advanced-3/pom.xml [hunk 5]
```diff
@@ -36,6 +67,30 @@
...
+                        <artifactId>aspectjtools</artifactId>
+                        <version>1.9.1</version>
+                    </dependency>
+                    <dependency>
+                        <groupId>org.aspectj</groupId>
+                        <artifactId>aspectjweaver</artifactId>
+                        <version>1.9.1</version>
+                    </dependency>
+                </dependencies>
+            </plugin>
```
[the remark this answers is not in the document]
Comment (SmartyAnsh) [reply core-java-modules/core-java-concurrency-advanced-3/pom.xml]: fixed

File: core-java-modules/core-java-concurrency-advanced-3/pom.xml (hunk 3 above)
Comment (SmartyAnsh) [reply core-java-modules/core-java-concurrency-advanced-3/pom.xml]: fixed

**Pull request merged**